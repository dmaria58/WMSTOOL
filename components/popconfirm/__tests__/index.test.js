import React from 'react';
import { mount } from 'enzyme';
import Popconfirm from '..';

describe('Popconfirm', () => {
  it('should popup Popconfirm dialog', () => {
    const onVisibleChange = jest.fn();

    const wrapper = mount(
      <Popconfirm
        title={<span className="popconfirm-test">Are you sure delete this task?</span>}
        okText="Yes"
        cancelText="No"
        mouseEnterDelay={0}
        mouseLeaveDelay={0}
        onVisibleChange={onVisibleChange}
      >
        <span>Delete</span>
      </Popconfirm>
    );

    const triggerNode = wrapper.find('span').at(0);
    triggerNode.simulate('click');
    expect(onVisibleChange).toBeCalledWith(true);
    expect(wrapper.find('.popconfirm-test').length).toBe(1);

    triggerNode.simulate('click');
    expect(onVisibleChange).toBeCalledWith(false);
  });

  it('should show overlay when trigger is clicked', () => {
    const popconfirm = mount(
      <Popconfirm title="code">
        <span>show me your code</span>
      </Popconfirm>
    );

    expect(popconfirm.instance().getPopupDomNode()).toBe(null);

    popconfirm.find('span').simulate('click');

    const popup = popconfirm.instance().getPopupDomNode();
    expect(popup).not.toBe(null);
    expect(popup.className).toContain('ant-popover-placement-top');
    expect(popup.innerHTML).toMatchSnapshot();
    expect(popup.innerHTML).toMatchSnapshot();
  });
});
