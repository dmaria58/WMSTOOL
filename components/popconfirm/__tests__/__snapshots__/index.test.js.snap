// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popconfirm should show overlay when trigger is clicked 1`] = `"<div class=\\"wmstool-popover-content\\"><div class=\\"wmstool-popover-arrow\\"></div><div class=\\"wmstool-popover-inner\\"><div><div class=\\"wmstool-popover-inner-content\\"><div class=\\"wmstool-popover-message\\"><i class=\\"anticon anticon-exclamation-circle\\"></i><div class=\\"wmstool-popover-message-title\\">code</div></div><div class=\\"wmstool-popover-buttons\\"><button type=\\"button\\" class=\\"wmstool-btn wmstool-btn-sm\\"><span>Cancel</span></button><button type=\\"button\\" class=\\"wmstool-btn wmstool-btn-primary wmstool-btn-sm\\"><span>OK</span></button></div></div></div></div></div>"`;

exports[`Popconfirm should show overlay when trigger is clicked 2`] = `"<div class=\\"wmstool-popover-content\\"><div class=\\"wmstool-popover-arrow\\"></div><div class=\\"wmstool-popover-inner\\"><div><div class=\\"wmstool-popover-inner-content\\"><div class=\\"wmstool-popover-message\\"><i class=\\"anticon anticon-exclamation-circle\\"></i><div class=\\"wmstool-popover-message-title\\">code</div></div><div class=\\"wmstool-popover-buttons\\"><button type=\\"button\\" class=\\"wmstool-btn wmstool-btn-sm\\"><span>Cancel</span></button><button type=\\"button\\" class=\\"wmstool-btn wmstool-btn-primary wmstool-btn-sm\\"><span>OK</span></button></div></div></div></div></div>"`;
